src/libraries/Unlocker.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.20;
+
+import {IHooks} from "../interfaces/IHooks.sol";
+
+/// @notice This is a temporary library that allows us to use transient storage (tstore/tload)
+/// TODO: This library can be deleted when we have the transient keyword support in solidity.
+library Unlocker {
+    // The slot holding the unlocker, transiently
+    uint256 constant UNLOCKER_SLOT = uint256(keccak256("Unlocker")) - 1;
+
+    /// @notice Thrown when trying to set the unlock address as address(0)
+    /// we use unlocker==address(0) to signal that the pool is not unlocked
+    error InvalidUnlocker();
+
+    function setUnlocked(address unlocker) internal {
```

Comment:
wondering if we only need to keep track of whether it's unlocked? what's the point of tracking the very first address that opened the lock?

Reply:
i agree dont see a point, but it costs no extra overhead also

Reply:
I made #487 to handle this in a new PR as this one is so big already